tasks/mrrc-collect-params/mrrc-collect-params.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: mrrc-collect-params
+spec:
+  params:
+    - name: dataJsonPath
+      description: path to data json file
+  results:
+    - name: mrrcParamFilePath
+      description: path of the env file for following tasks
+    - name: cosignPubKeyFile
+      description: the public key file name for cosign verification
+    - name: cosignPubKeySecret
+      description: the secret name for cosign verify public key
+    - name: charonAWSSecret
+      description: the secret name for charon aws credential file
+  steps:
+    - name: mrrc-collect-params
+      image: quay.io/konflux-ci/yq:latest
+      script: |
+        set -eux
+
+        cd $(workspaces.data.path)
+        mrrcEnvFile="./mrrc.env"
+        if [ -f $mrrcEnvFile ]; then
+          rm $mrrcEnvFile
+        fi
+
+        zipRegistry=$(yq -r '.zipRegistry' $(params.dataJsonPath))
+        echo "export MRRC_ZIP_REGISTRY=$zipRegistry" >> $mrrcEnvFile
+
+        environment=$(yq -r '.mrrc.environment' $(params.dataJsonPath))
+        release=$(yq -r '.mrrc.release' $(params.dataJsonPath))
+        target=$environment-maven-$release
+        echo "export MRRC_TARGET=$target" >> $mrrcEnvFile
+
+        productName=$(yq -r '.mrrc.product.name' $(params.dataJsonPath))
+        productVersion=$(yq -r '.mrrc.product.version' $(params.dataJsonPath))
+        echo "export MRRC_PRODUCT_NAME=$productName" >> $mrrcEnvFile
+        echo "export MRRC_PRODUCT_VERSION=$productVersion" >> $mrrcEnvFile
+
+        awsSecret=$(yq -r '.mrrc.awsSecret' $(params.dataJsonPath))
+        echo -n "$awsSecret" > "$(results.charonAWSSecret.path)"
+
+        consignPubKey=$(yq -r '.cosignPubKeyFile' $(params.dataJsonPath))
```

Comment:
Shouldn't it be `cosignPubKey` not `consignPubKey`?

Reply:
Oh sorry for the typo.

Reply:
> Why use the snapshotSpec instead of the data file which is already a parameter to this task?

Please see answer here: https://github.com/konflux-ci/release-service-catalog/pull/493#discussion_r1696118803

Reply:
Fixed in new commit